### DISCUSSION TITLE
It is not possible to set a property on a view UserControl inside another UserControl

### DISCUSSION THREAD

**Discussion by HavenDV** [unoplatform/uno#8057] [General]
I am trying to set the Background property of a SettingsView inside another View. Uno throws an error when compiling:
```
1>Uno.UI.SourceGenerators\Uno.UI.SourceGenerators.XamlGenerator.XamlCodeGenerator\LoginView_b954beae61eca716b6d97c6bd062df79.cs(738,34,738,244): error CS0266: Cannot implicitly convert type 'object' to 'Windows.UI.Xaml.Media.Brush'. An explicit conversion exists (are you missing a cast?)
```

```xml
<views:SettingsView
    Background="{ThemeResource AppBarBackgroundThemeBrush}"
    />
```
where SettingsView:
```xml
<UserControl
    x:Class="Views.SettingsView"
    >
```

I'm sure a related issue already exists somewhere in Uno, so I started with a discussion

**Comment by jeromelaban** [comment]
Thanks for the report. This may look like a Roslyn-related issue where the base type cannot be determined. Is `SettingsViewBase` source-generated in any way ?

**Reply by HavenDV** [reply]
I tested both the generated variant and not. Both give the same error. The important point is that SettingsViewBase is inside another library.

**Reply by jeromelaban** [reply]
Thanks. Does a "normal" `UserControl` cause the same issue?

**Reply by HavenDV** [reply]
```cs
public abstract partial class SettingsViewBase : UserControl
{
}
```
causes the same issue.

**Reply by HavenDV** [reply]
Even just 
```xml
<UserControl
    x:Class="Views.SettingsView"
    >
```
located in same library causes this problem

**Reply by HavenDV** [reply]
Do I need to convert this to an issue?

**Reply by jeromelaban** [reply]
Yes, with a repro case as it seems non-trivial.
